### PR TITLE
Fix building static libs

### PR DESCRIPTION
**Pull request by m-kuhn**
Port of https://github.com/microsoft/vcpkg/pull/17506

### CODE REVIEW

**Comment by m-kuhn** [comment]
I don't understand the connection between the changes in here and the c++20 build errors. :thinking:

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -204,11 +208,12 @@ if(MSVC)
...
-  target_compile_definitions(spatialindex PRIVATE -DSIDX_DLL_EXPORT=1)
-  if(NOT WITH_STATIC_SIDX)
-    target_compile_definitions(spatialindex PRIVATE -DSIDX_DLL_IMPORT=1)
-    target_compile_definitions(spatialindex_c PRIVATE -DSIDX_DLL_IMPORT=1)
+ if (BUILD_SHARED_LIBS)
+   target_compile_definitions(spatialindex PRIVATE -DSIDX_DLL_EXPORT=1)
+   target_compile_definitions(spatialindex_c PRIVATE -DSIDX_DLL_EXPORT=1)
+ else()
+  target_compile_definitions(spatialindex PRIVATE -DSIDX_DLL_IMPORT=1)
+  target_compile_definitions(spatialindex_c PRIVATE -DSIDX_DLL_IMPORT=1)
```
Comment (mwtoews) [inline src/CMakeLists.txt:216]: Awesome to remove custom `WITH_STATIC_SIDX`, as a global `BUILD_SHARED_LIBS` is preferred.

Two minor nits about style: no space after the `if` command, and use a multiple of two space indents (not one).

**Review by mwtoews — CHANGES_REQUESTED** [review]
Here are a few suggested edits. Other changes are welcome!

As for the C++20 errors, these are Clang issues with their `-std=c++2a` partial C++20 support.

**Comment by m-kuhn** [comment]
Thanks for the suggestions, pr adjusted.
Do the header changes look good to you?

**Review by mwtoews — APPROVED** [review]
Thanks, looks good!

Static Windows builds should eventually be brought into the CI jobs.

**Comment by m-kuhn** [comment]
> Static Windows builds should eventually be brought into the CI jobs.

Tried to do sobut it doesn't appear yet. Does it only run when it's on master?

**Comment by hobu** [comment]
I have refactored away from AzP to use GitHub Actions. Please update this PR to catch up and get it ✅ if you are interested in it be including in the next release.

**Comment by m-kuhn** [comment]
I am away from a keyboard until June, so I am afraid I cannot do a lot this time. If someone could merge master into this branch I would at least have the gh workflow file to edit around, the patch should be trivial enough to do on a mobile device.

**Comment by hobu** [comment]
> I am away from a keyboard until June, so I am afraid I cannot do a lot this time. 

It's ok. It is very unlikely there will be a release by June anyway 😄  We will leave this open and someone can pick it up.

**Comment by m-kuhn** [comment]
I tried to add the required github workflow matrix magic, fingers crossed that's how it works

**Comment by hobu** [comment]
CI failing https://github.com/libspatialindex/libspatialindex/actions/runs/2492840442

**Pull request merged**